libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2303,11 +2303,22 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 		functionType = dynamic_cast<FunctionType const*>(expressionType);
 		funcCallAnno.kind = FunctionCallKind::FunctionCall;
 
+		{
+			FunctionDefinition const* expressionDeclaration = nullptr;
```

Comment:
Why the complication with this local variable?

Reply:
I tried to avoid having the `calledDirectly = true` part repeated..  will chaneg